src/savegame_json.cpp

```diff
@@ -1353,6 +1353,34 @@ void item::io( Archive& archive )
         // only for backward compatibility (nowadays mode is stored in item_vars)
         set_gun_mode( mode );
     }
+
+    if( !magazine_integral() ) {
+        // items with detachable magazines shouldn't have charges or curammo set
+        if( curammo || charges != 0 ) {
+            if( curammo && charges > 0 ) {
+                item mag( type->magazine_default, calendar::turn );
+                item ammo( curammo->id, calendar::turn );
+
+                ammo.charges = std::min( charges, mag.ammo_capacity() );
+                mag.contents.push_back( ammo );
+
+                contents.push_back( mag );
+            }
+            charges = 0;
+            curammo = nullptr;
+        }
+
+        // remove incompatible magazine mods
+        if( has_gunmod( "spare_mag" ) >= 0 ) {
+            contents.erase( contents.begin() + has_gunmod( "spare_mag" ) );
```

Comment:
> Hmm, do we not know where the gun is on the map?

From this function I don't think so?

It's not that it can't be done it's just that it's probably considerably more complex.

I'm also still uncertain as to how to distinguish between guns that have been converted (`charges == 0, curammo == nullptr`) and those that haven't but currently have no ammo. If a player already has an empty gun in the inventory they really should get an empty magazine to go with it as part of this conversion? The only solution I can think of is a `legacy` flag or equivalent attached to the gun applied after conversion?